### PR TITLE
Added the DoIT ESPduino-32 board

### PR DESCRIPTION
**Pull request by alexceltare2**
Some pins missing but will finish as soon as I'm done researching it.

### CODE REVIEW

**Comment by alexceltare2** [comment]
IS someone going to approve the edit?

File: variants/doitESPduino32/pins_Arduino.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+static const uint8_t IO14 = 14;
+static const uint8_t IO27 = 27;
+static const uint8_t IO16 = 16;
+static const uint8_t IO17 = 17;
+static const uint8_t IO25 = 25;
+static const uint8_t IO26 = 26;
+static const uint8_t TX0 = 1;
+static const uint8_t RX0 = 3;
+//SD
+static const uint8_t SD2 = 9;
```
[the remark this answers is not in the document]
Comment (alexceltare2) [reply variants/doitESPduino32/pins_Arduino.h]: Yeah, but on the board they are labelled as SD2, SD3, CMD..., as you can see here (http://img.dxcdn.com/productimages/sku_474179_1.jpg). These pins are not defined and the developer doesn't know it until he tries out the pin numbers.

File: variants/doitESPduino32/pins_Arduino.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+static const uint8_t LED_BUILTIN = 2;
+#define BUILTIN_LED  LED_BUILTIN // backward compatibility
+
+
+static const uint8_t SDA = 21;
+static const uint8_t SCL = 22;
+
+//SPI
+static const uint8_t IO5    = 5;
```
[the remark this answers is not in the document]
Comment (alexceltare2) [reply variants/doitESPduino32/pins_Arduino.h]: done

**Comment by me-no-dev** [comment]
hmm why are there merge conflicts? seems like some of my commits are also included?

**Comment by alexceltare2** [comment]
Alright Mr. me-no-dev. I've updated all it needs and followed the doitESP32devkitV1 model. Also, I've kept the boards.txt as the original as github seems to find conflicts if I change it.

**Comment by atanisoft** [comment]
this is no longer necessary for merge as the d-duino-32 board is identical (and is the actual board that the DoIT ESPduino-32 board cloned)

**Comment by alexceltare2** [comment]
> this is no longer necessary for merge as the d-duino-32 board is identical (and is the actual board that the DoIT ESPduino-32 board cloned)

WTF? d-duino-32 is a oled attached mini board with limited pin numbers whereas ESPduino-32 is a fully featured board what all pins exposed. Not even the pin markings are the same so your argument is invalid.

**Comment by atanisoft** [comment]
> d-duino-32 is a oled attached mini board with limited pin numbers whereas ESPduino-32

Sorry you are right, I did a quick search at the time and it came up as the same. Now looking again it is definitely different.

However, there are some pin mappings that are not correct on this board. A0 and A1 are mapped oddly to 2 and 4 so your pins_arduino.h should reflect this order on board (maybe even expose the A? constants?) I also don't see GPIO 0 exposed anywhere, where did you find it on the board?

**Review by me-no-dev — APPROVED** [review]

**Pull request merged**